### PR TITLE
fix(core): delete some of the redundant code in core

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: packages/core/src/LoadNodeListSearch.ts [hunk 1]
```diff
@@ -1,97 +1,11 @@
-/* eslint-disable no-restricted-syntax */
-/* eslint-disable @typescript-eslint/no-unsafe-return */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
-/* eslint-disable @typescript-eslint/restrict-template-expressions */
-/* eslint-disable @typescript-eslint/no-unsafe-call */
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { INodeListSearchResult, IWorkflowExecuteAdditionalData } from 'n8n-workflow';
```
Comment (ivov) [inline packages/core/src/LoadNodeListSearch.ts:1]:
```suggestion
import type { INodeListSearchResult, IWorkflowExecuteAdditionalData } from 'n8n-workflow';
```

File: packages/core/src/LoadNodeDetails.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import {
+	INode,
+	INodeCredentials,
+	INodeParameters,
+	INodeTypeNameVersion,
+	INodeTypes,
+	Workflow,
+} from 'n8n-workflow';
```
Comment (ivov) [inline packages/core/src/LoadNodeDetails.ts:8]: Is there a benefit in splitting out type imports from runtime imports?

```ts
import type {
	INode,
	INodeCredentials,
	INodeParameters,
	INodeTypeNameVersion,
	INodeTypes,
} from 'n8n-workflow';
import { Workflow } from 'n8n-workflow';
```

File: packages/core/src/LoadNodeListSearch.ts [hunk 3]
```diff
@@ -1,97 +1,11 @@
...
-			name: TEMP_WORKFLOW_NAME,
-			active: false,
-			connections: {},
-			nodes: Object.values(this.workflow.nodes),
-			createdAt: new Date(),
-			updatedAt: new Date(),
-		};
-	}
 
+export class LoadNodeListSearch extends LoadNodeDetails {
```
Comment (ivov) [inline packages/core/src/LoadNodeListSearch.ts:8]: I thought we were getting rid of JSDoc?

**Review by ivov — DISMISSED** [review]
Tested `methods.loadOptions` with Baserow and `methods.listSearch` with GDrive.

File: packages/core/src/LoadNodeParameterOptions.ts [hunk 4]
```diff
@@ -106,29 +25,26 @@ export class LoadNodeParameterOptions {
 		methodName: string,
 		additionalData: IWorkflowExecuteAdditionalData,
 	): Promise<INodePropertyOptions[]> {
-		const node = this.workflow.getNode(TEMP_NODE_NAME);
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const node = this.workflow.getNode(TEMP_NODE_NAME)!;
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/core/src/LoadNodeParameterOptions.ts]: I instead moved this line into `getTempNode()` in the base class. that way `TEMP_NODE_NAME` is only defined and used in a single place.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.200.0`

**Pull request merged**